src/components/AssetSelection/AssetSelection.tsx

```diff
@@ -34,16 +34,40 @@ type TradeAssetSelectEditableProps = {
 
 type TradeAssetSelectProps = TradeAssetSelectReadonlyProps | TradeAssetSelectEditableProps
 
-export const TradeAssetSelect: React.FC<TradeAssetSelectProps> = ({
-  onAssetClick,
-  onAssetChange,
-  assetId,
-  assetIds,
-  isReadOnly,
-  isLoading,
-  buttonProps,
-  ...rest
-}) => {
+export const TradeAssetSelect: React.FC<TradeAssetSelectProps> = memo(props => {
+  const {
+    onAssetClick,
+    onAssetChange,
+    assetId,
+    assetIds,
+    isReadOnly,
+    isLoading,
+    buttonProps,
+    flexProps,
+  } = useMemo(() => {
+    const {
+      onAssetClick,
+      onAssetChange,
+      assetId,
+      assetIds,
+      isReadOnly,
+      isLoading,
+      buttonProps,
+      ...flexProps
+    } = props
+    return {
+      onAssetClick,
+      onAssetChange,
+      assetId,
+      assetIds,
+      isReadOnly,
+      isLoading,
+      buttonProps,
+      flexProps,
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [Object.values(props)])
```

Comment:
Not sure about this guy:

https://react.dev/reference/react/useMemo

`React will compare each dependency with its previous value using the [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison.`

However, `Object.is(Object.values({}), Object.values({})) === false`, since these are effectively two different references.

Reply:
If we want to achieve stable references, perhaps stable JSON stringify is our guy? Since `Object.is` will compare two stable strings, i.e primitives, and will return `true`

Reply:
fixed in a0ccb23533097549f33f75f7d537da43ed9d690b 🙏